File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/solution/unary_plus.cpp
// unary_plus.cpp

#include <iostream>
#include <string>
#include "dp_exception.h"
#include "dp_map.h"
class expression;
#include "unary.h"
#include "unary_plus.h"

unary_plus::unary_plus(expression * a_operand1)
: unary(a_operand1)
{
}

double unary_plus::get_value(const dp_map<std::string, double> & a_map) const
{
	if (get_operand1() == nullptr)
	{
		throw dp_exception("double unary_plus::get_value(dp_map<std::string, double> & a_map) const -- get_operand1() == nullptr.");
	}
	double a_value1 = get_operand1()->get_value(a_map);

	double result(+ a_value1);

	return result;
}

void unary_plus::output(std::ostream & a_ostream) const
{
	output_prefix(a_ostream, "+");
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/solution/UnaryPlus.java
// UnaryPlus.java

import java.io.PrintWriter;

public class UnaryPlus extends Unary
{
	public UnaryPlus(Expression aOperand1)
	{
		super(aOperand1);
	}

	public double getValue(DpMap<String, Double> aMap)
	{
		if (getOperand1() == null)
		{
			throw new DpException("void UnaryPlus.getValue(DpMap<String, Double> aMap) -- getOperand1() == null.");
		}
		double aValue1 = getOperand1().getValue(aMap);

		double result = + aValue1;

		return result;
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputPrefix(aPrintWriter, "+");
	}
}